eo-runtime/src/main/java/org/eolang/PhTracedEnclosure.java

```diff
@@ -132,16 +158,64 @@ private TracingWhileGetting(final Supplier<Attr> attr) {
 
         @Override
         public Attr get() {
-            if (PhTracedEnclosure.DATAIZING_CAGES.contains(PhTracedEnclosure.this.cage)) {
-                throw new ExFailure(
-                    "The cage %s is already dataizing",
+            final Integer incremented = this.incrementCageCounter();
+            final Attr ret = this.attr.get();
+            this.decrementCageCounter(incremented);
+            return ret;
+        }
+
+        /**
+         * Increments counter of cage in the {@link PhTracedEnclosure#DATAIZING_CAGES}.
+         * @return New value in the map.
+         */
+        private Integer incrementCageCounter() {
+            return PhTracedEnclosure.DATAIZING_CAGES.compute(
+                PhTracedEnclosure.this.cage, (key, counter) -> {
+                    final int ret = this.incremented(counter);
+                    if (ret > PhTracedEnclosure.this.depth) {
+                        throw new ExFailure(
+                            "The cage %s is already dataizing",
```

Comment:
@levBagryansky the message is not clear. Imagine you're end-user and you know nothing about the cage pitfall we're trying to prevent here. And you get such message after your program failed. What does it mean? What should I do and so on...